From 462aa57a994b2db1304a945b11de483bc2ecbb2d Mon Sep 17 00:00:00 2001
From: Ong Boon Leong <boon.leong.ong@intel.com>
Date: Wed, 3 Jun 2020 00:09:17 +0800
Subject: [PATCH 38/54] net: stmmac: fix IRQ name issue in TX XDP Queue during
 tear-down

During tear down, we need to change "tx-xdp-N" to "tx-N", so we
need to pick up the right name based on return of queue_is_xdp()

Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index b33ccfa30a3f..1ae7bea056d4 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -5621,8 +5621,12 @@ static int stmmac_txrx_irq_control(struct stmmac_priv *priv, u16 qid, bool en)
 			goto irq_err;
 
 		int_name = priv->int_name_tx_irq[qid + qp_num];
-		sprintf(int_name, "%s:%s-%d", priv->dev->name,
-			"tx-xdp", qid + qp_num);
+		if (queue_is_xdp(priv, qid + qp_num))
+			sprintf(int_name, "%s:%s-%d", priv->dev->name,
+				"tx-xdp", qid + qp_num);
+		else
+			sprintf(int_name, "%s:%s-%d", priv->dev->name,
+				"tx", qid + qp_num);
 		ret = request_irq(priv->tx_irq[qid + qp_num],
 				  stmmac_msi_intr_tx,
 				  0, int_name,
-- 
2.17.0

